include/Board.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "Piece.h"
+#include "macros.h"
+#include "Factory.h"
+#include <memory>
+#include <iostream>
+
+
+
+class Board {
+public:
+	Board(const std::string& board);
```

Comment:
consider using const where appropriate for member functions that do not modify the state of the object.